sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/common/ImmutableDoubleArray.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.common;
+
+import java.util.Arrays;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class ImmutableDoubleArray {
+  private static final ImmutableDoubleArray EMPTY = new ImmutableDoubleArray(new double[0]);
+
+  /** Returns the empty array. */
+  public static ImmutableDoubleArray of() {
```

Comment:
do we need this method? let's keep the API as narrow as possible for now. Please remove methods that we don't need. thanks!

Reply:
this is required to construct an empty histogram accumulation when merging two accumulation with different boundaries. I won't argue that it's strictly needed though.

Reply:
I think we should remove this as well as the boundaries from the Accumulation.